From ad4c3afa0de51e15e295bd3edcb3516f7b2169e4 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:27 -0400
Subject: [powerpc] book3s: Fix some ABIv2 issues in machine check code

Message-id: <1410545655-205645-579-git-send-email-dzickus@redhat.com>
Patchwork-id: 94522
O-Subject: [RHEL7 PATCH 578/626] powerpc/book3s: Fix some ABIv2 issues in machine check code
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit ad718622ab6d500c870772b1b8dda46fa2195e6d
Author: Anton Blanchard <anton@samba.org>
Date:   Thu Jun 12 09:08:37 2014 +1000

    powerpc/book3s: Fix some ABIv2 issues in machine check code

    Commit 2749a2f26a7c (powerpc/book3s: Fix machine check handling for
    unhandled errors) introduced a few ABIv2 issues.

    We can maintain ABIv1 and ABIv2 compatibility by branching to the
    function rather than the dot symbol.

    Fixes: 2749a2f26a7c ("powerpc/book3s: Fix machine check handling for unhandled errors")
    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/exceptions-64s.S b/arch/powerpc/kernel/exceptions-64s.S
index 8a0292d..a7d36b1 100644
--- a/arch/powerpc/kernel/exceptions-64s.S
+++ b/arch/powerpc/kernel/exceptions-64s.S
@@ -1502,13 +1502,13 @@ machine_check_handle_early:
 unrecover_mce:
  /* Invoke machine_check_exception to print MCE event and panic. */
  addi r3,r1,STACK_FRAME_OVERHEAD
- bl .machine_check_exception
+ bl machine_check_exception
  /*
   * We will not reach here. Even if we did, there is no way out. Call
   * unrecoverable_exception and die.
   */
 1: addi r3,r1,STACK_FRAME_OVERHEAD
- bl .unrecoverable_exception
+ bl unrecoverable_exception
  b 1b
 /*
  * r13 points to the PACA, r9 contains the saved CR,
-- 
1.7.1